File: backend/src/models/Discovery.js
module.exports = (sequelize, DataTypes) => {
  const Discovery = sequelize.define('Discovery', {
    id: {
      type: DataTypes.UUID,
      defaultValue: DataTypes.UUIDV4,
      primaryKey: true
    },
    treasure_id: {
      type: DataTypes.UUID,
      allowNull: false,
      references: {
        model: 'treasures',
        key: 'id'
      }
    },
    user_id: {
      type: DataTypes.UUID,
      allowNull: false,
      references: {
        model: 'users',
        key: 'id'
      }
    },
    discovered_at: {
      type: DataTypes.DATE,
      defaultValue: DataTypes.NOW
    },
    points_earned: {
      type: DataTypes.INTEGER,
      defaultValue: 0
    },
    time_to_find: {
      type: DataTypes.INTEGER,
      allowNull: true,
      comment: 'Time in seconds from treasure creation to discovery'
    },
    distance_from_treasure: {
      type: DataTypes.FLOAT,
      allowNull: true,
      comment: 'Distance in meters when discovered'
    },
    photo_url: {
      type: DataTypes.STRING,
      allowNull: true
    },
    comment: {
      type: DataTypes.TEXT,
      allowNull: true
    },
    reaction_type: {
      type: DataTypes.ENUM('like', 'love', 'wow', 'funny', 'cool'),
      allowNull: true
    }
  }, {
    tableName: 'discoveries',
    timestamps: true,
    underscored: true,
    indexes: [
      {
        unique: true,
        fields: ['treasure_id', 'user_id']
      }
    ]
  });

  return Discovery;
};

File: didactic-barnicle-ios/didactic-barnicle-ios/Models/Discovery.swift
import Foundation

enum ReactionType: String, Codable, CaseIterable {
    case like = "like"
    case love = "love"
    case wow = "wow"
    case funny = "funny"
    case cool = "cool"
}

struct Discovery: Codable, Identifiable {
    let id: UUID
    let treasureId: UUID
    let userId: UUID
    let discoveredAt: Date
    var photoURL: String?
    var comment: String?
    var reactionType: ReactionType?
    var pointsEarned: Int
    var timeToFind: Int?
    var distanceFromTreasure: Float?
    
    init(
        id: UUID = UUID(),
        treasureId: UUID,
        userId: UUID,
        discoveredAt: Date = Date(),
        photoURL: String? = nil,
        comment: String? = nil,
        reactionType: ReactionType? = nil,
        pointsEarned: Int = 0,
        timeToFind: Int? = nil,
        distanceFromTreasure: Float? = nil
    ) {
        self.id = id
        self.treasureId = treasureId
        self.userId = userId
        self.discoveredAt = discoveredAt
        self.photoURL = photoURL
        self.comment = comment
        self.reactionType = reactionType
        self.pointsEarned = pointsEarned
        self.timeToFind = timeToFind
        self.distanceFromTreasure = distanceFromTreasure
    }
}